src/components/featureHighlight/index.js

```diff
@@ -148,11 +148,19 @@ class FeatureHighlight extends BaseComponent {
     this.setTargetPosition();
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.setTargetPosition(nextProps);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const target = nextProps?.getTarget?.();
+    const node = FeatureHighlight.findTargetNode(target);
+    if (node && node !== prevState?.node) {
+      return {...prevState, node};
+    }
+    return null;
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(nextProps) {
+    if (!_.isEqual(nextProps, this.props)) {
```

Comment:
Is it really necessary to check all the props? All of them are relevant for setting the target position?

Reply:
Right, fixed